test/services/licences/supplementary/create-licence-supplementary-year.service.test.js

```diff
@@ -0,0 +1,93 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+
+const { describe, it, beforeEach } = exports.lab = Lab.script()
+const { expect } = Code
+
+// Test helpers
+const { generateUUID } = require('../../../../app/lib/general.lib.js')
+const LicenceSupplementaryYearHelper = require('../../../support/helpers/licence-supplementary-year.helper.js')
+const LicenceSupplementaryYearModel = require('../../../../app/models/licence-supplementary-year.model.js')
+
+// Thing under test
+const CreateLicenceSupplementaryYearService = require('../../../../app/services/licences/supplementary/create-licence-supplementary-year.service.js')
+
+describe('Create Licence Supplementary Years Service', () => {
+  let licenceId
+  let twoPartTariff
+  let years
+
+  describe('when provided a licenceId, years and twoPartTariff data', () => {
+    beforeEach(async () => {
+      licenceId = generateUUID()
+      twoPartTariff = true
+      years = [2023]
+    })
+
+    describe('that does not already exist', () => {
+      it('persists the data', async () => {
+        await CreateLicenceSupplementaryYearService.go(licenceId, years, twoPartTariff)
+
+        const result = await _fetchLicenceSupplementaryYears(licenceId)
+
+        expect(result).to.have.length(1)
+        expect(result[0].licenceId).to.equal(licenceId)
```

Comment:
I would suggest checking all the fields. Else, we have nothing confirming the year got persisted (I think) in these tests.

```suggestion
        expect(result[0]).to.equal({
          licenceId,
          billRunId: null,
          financialYearEnd: years[0],
          twoPartTariff
        }, { skip: ['id'] })
```